greenheart/tools/optimization/gc_PoseOptimization.py

```diff
@@ -0,0 +1,505 @@
+            n_DV *= 2
+
+        return n_DV
+
+    def _get_step_size(self):
+        """If a step size for the driver-level finite differencing is provided, use that step size. Otherwise use a default value.
+
+        Returns:
+            step size (float): step size for optimization
+        """
+
+        return (
+            1.0e-6
+            if not "step_size" in self.config.greenheart_config["opt_options"]["driver"]["optimization"]
+            else self.config.greenheart_config["opt_options"]["driver"]["optimization"]["step_size"]
+        )
+
+    def _set_optimizer_properties(self, opt_prob, options_keys=[], opt_settings_keys=[], mapped_keys={}):
+        """
+        Set the optimizer properties, both the `driver.options` and
+        `driver.opt_settings`. See OpenMDAO documentation on drivers
+        to determine which settings are set by either options or
+        opt_settings.
+
+        Parameters
+        ----------
+        opt_prob : OpenMDAO problem object
+            The wind turbine problem object.
+        options_keys : list
+            List of keys for driver options to be set.
+        opt_settings_keys: list
+            List of keys for driver opt_settings to be set.
+        mapped_keys: dict
+            Key pairs where the yaml name differs from what's expected
+            by the driver. Specifically, the key is what's given in the yaml
+            and the value is what's expected by the driver.
+
+        Returns
+        -------
+        opt_prob : OpenMDAO problem object
+            The updated wind turbine problem object with driver settings applied.
+        """
+
+        opt_options = self.config.greenheart_config["opt_options"]["driver"]["optimization"]
+
+        # Loop through all of the options provided and set them in the OM driver object
+        for key in options_keys:
+            if key in opt_options:
+                if key in mapped_keys:
+                    opt_prob.driver.options[mapped_keys[key]] = opt_options[key]
+                else:
+                    opt_prob.driver.options[key] = opt_options[key]
+
+        # Loop through all of the opt_settings provided and set them in the OM driver object
+        for key in opt_settings_keys:
+            if key in opt_options:
+                if key in mapped_keys:
+                    opt_prob.driver.opt_settings[mapped_keys[key]] = opt_options[key]
+                else:
+                    opt_prob.driver.opt_settings[key] = opt_options[key]
+
+        return opt_prob
+
+    def set_driver(self, opt_prob):
+        """set which optimization driver to use and set options
+
+        Args:
+            opt_prob (openmdao problem instance): openmdao problem class instance for current optimization problem
+
+        Raises:
+            ImportError: _description_
```

Comment:
Add descriptions to Raises Errors or remove section from docstring

Reply:
Oops, thanks. Fixed